xdsl/ir/core.py

```diff
@@ -50,13 +52,18 @@ def operations(self) -> Iterator[type[Operation]]:
     def attributes(self) -> Iterator[type[Attribute]]:
         return iter(self._attributes)
 
+    @property
+    def name(self) -> str:
+        return self._name
+
 
 @dataclass
 class MLContext:
     """Contains structures for operations/attributes registration."""
 
     allow_unregistered: bool = field(default=False)
 
+    _loaded_dialects: dict[str, Dialect] = field(init=False, default_factory=dict)
     _loaded_ops: dict[str, type[Operation]] = field(init=False, default_factory=dict)
     _loaded_attrs: dict[str, type[Attribute]] = field(init=False, default_factory=dict)
```

Comment:
We could also not load the operations/attrs individually and go through all the dialects when looking up. What do you think?

Reply:
I like it in general, but it requires some design and thoughts, especially wrt to unregistered attributes and operations

Would you be comfortable merging as is, and starting a discussion over Zulip? I do think it's valuable to think about it, and at the same time I'd like to enable whatever just naming dialects enables first 🙂 (opaque dialect attributes being my first target)

Reply:
Sure, this is essentially a first "backwards compatible" version that saves us from having to rename all oeprations, so I am fine with first merging this.

Reply:
Exactly, cool :slightly_smiling_face: